src/lib/optimizer/optimizer.cpp

```diff
@@ -17,12 +18,14 @@ Optimizer Optimizer::create_default_optimizer() {
   Optimizer optimizer{10};
 
   RuleBatch main_batch(RuleBatchExecutionPolicy::Iterative);
-
   main_batch.add_rule(std::make_shared<PredicateReorderingRule>());
   main_batch.add_rule(std::make_shared<JoinDetectionRule>());
-
   optimizer.add_rule_batch(main_batch);
 
+  RuleBatch final_batch(RuleBatchExecutionPolicy::Once);
```

Comment:
`main_batch` and `final_batch` seem to be quite useless names to me. Maybe we need to looks at the interface again so that we don't have to store the batches in variables.

Reply:
Okay, but should we stick with it for now?

Reply:
We don't have to make that part of this PR